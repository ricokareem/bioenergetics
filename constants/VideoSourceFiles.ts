/* eslint-disable global-require */

const VideoSourceFiles = {
  file1: require("../media/1.mp4"),
  file2: require("../media/2.mp4"),
  file3: require("../media/3.mp4"),
  file4: require("../media/4.mp4"),
  file5: require("../media/5.mp4"),
  file6: require("../media/6.mp4"),
  file7: require("../media/7.mp4"),
  file9: require("../media/9.mp4"),
  file10: require("../media/10.mp4"),
  file11: require("../media/11.mp4"),
  file81: require("../media/81.mp4"),
  file82: require("../media/82.mp4"),
  file121: require("../media/121.mp4"),
  file122: require("../media/122.mp4"),
  file123: require("../media/123.mp4"),
  file131: require("../media/131.mp4"),
  file132: require("../media/132.mp4"),
  file133: require("../media/133.mp4"),
};

export default VideoSourceFiles;
