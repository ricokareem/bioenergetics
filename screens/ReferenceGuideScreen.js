import React from 'react';
import PropTypes from 'prop-types';
import {
  ScrollView,
  StyleSheet,
  View
} from 'react-native';
import { ListItem } from 'react-native-elements';

const symptoms = [
  {name: 'Cards', title: 'MIGRAINE/HEADACHES'},
  {name: 'Cards', title: 'NECK PAIN'},
  {name: 'Cards', title: 'OVERALL BODY PAIN'},
  {name: 'Cards', title: 'GASTRIC/STOMACH PAIN/SPASMS'},
  {name: 'Cards', title: 'DIARRHEA'},
  {name: 'Cards', title: 'CHRONIC CONSTIPATION'},
  {name: 'Cards', title: 'FLU'},
  {name: 'Cards', title: 'BLADDER PAIN'},
  {name: 'Cards', title: 'PMS/CRAMPS'},
  {name: 'Cards', title: 'DEMENTIA'},
  {name: 'Cards', title: 'THYROID'},
  {name: 'Cards', title: 'KNEE PAIN'},
  {name: 'Cards', title: 'LEG PAIN/CRAMPS'},
];

const chevronProps = {
  color: "#fff",
  size: 28,
}

const ReferenceGuideScreen = (props) => {
  // static navigationOptions = {
  //   title: 'Links',
  // };

  return (
    <View style={styles.container}>
      <ScrollView>
        {symptoms.map((item, index) => (
          <ListItem
            key={index}
            title={item.title}
            containerStyle={styles.listItem}
            titleStyle={styles.listText}
            onPress={() => navigation.navigate(item.name)}
            bottomDivider
            chevron={chevronProps}
          />
        ))}
      </ScrollView>
    </View>
  );
}

ReferenceGuideScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fc9829',
  },
  listItem: {
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    backgroundColor: '#fc9829',
  },
  listText: {
    fontSize: 24,
    lineHeight: 24,
    color: '#fff',
  },
});

export default ReferenceGuideScreen;
