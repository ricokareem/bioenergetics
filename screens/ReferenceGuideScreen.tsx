import React, { ReactElement } from "react";
import { Route, ScrollView, StyleSheet, View } from "react-native";
import { ListItem } from "react-native-elements";
import { SymptomsPlaylists } from "../constants/MediaData";

type NavPropsType = {
  navigation: Route;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fc9829",
  },
  listText: {
    fontSize: 14,
    fontWeight: "bold",
    lineHeight: 14,
    color: "#fff",
  },
});

function ReferenceGuideScreen({ navigation }: NavPropsType): ReactElement {
  return (
    <View style={styles.container}>
      <ScrollView>
        {SymptomsPlaylists.map((item) => (
          <ListItem
            key={item.id}
            containerStyle={{
              justifyContent: "space-between",
              alignItems: "center",
              borderBottomWidth: 1,
              borderBottomColor: "#ccc",
              backgroundColor: "#fc9829",
            }}
            onPress={() =>
              navigation.navigate(item.screen, {
                title: item.title,
                playlist: item.movieIds,
              })
            }
            bottomDivider
          >
            <ListItem.Title style={styles.listText}>
              {item.title}
            </ListItem.Title>
            <ListItem.Chevron color="#fff" size={28} />
          </ListItem>
        ))}
      </ScrollView>
    </View>
  );
}

export default ReferenceGuideScreen;
