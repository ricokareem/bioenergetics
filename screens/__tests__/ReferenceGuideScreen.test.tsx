import React from "react";
import { render } from "@testing-library/react-native";
import ReferenceGuideScreen from "../ReferenceGuideScreen";
import { SymptomsPlaylists } from "../../constants/MediaData";

describe("<ReferenceGuideScreen>", () => {
  it("should match snapshot", () => {
    const { toJSON } = render(
      <ReferenceGuideScreen navigation={{ id: "test" }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render all ListItem buttons", () => {
    const { getAllByTestId, getByText } = render(
      <ReferenceGuideScreen navigation={{ id: "test" }} />
    );
    const ListItemButtons = getAllByTestId("padView");
    const titles = SymptomsPlaylists.map((playlist) => playlist.title);

    expect(ListItemButtons).toHaveLength(15);
    titles.forEach((title) => {
      expect(getByText(title));
    });
  });
});

// LINKS COVERED IN CYPRESS TEST
