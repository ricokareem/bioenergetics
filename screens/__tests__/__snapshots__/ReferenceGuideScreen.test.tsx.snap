// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReferenceGuideScreen> should match to snapshot 1`] = `ShallowWrapper {}`;
